sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/aggregate/percentiles.scala

```diff
@@ -59,7 +59,8 @@ abstract class PercentileBase
 
   override lazy val dataType: DataType = {
     val resultType = child.dataType match {
-      case it: AnsiIntervalType => it
+      case _: YearMonthIntervalType => YearMonthIntervalType()
+      case _: DayTimeIntervalType => DayTimeIntervalType()
```

Comment:
shall we still respect the start field? e.g. `median(interval minute)` returns internal minute to second`.

Reply:
It could make sense, but the current approach is we return the default type (interval day to second) in many places. See `mean()` (PR's description), for instance. cc @srielau

Reply:
Ah I see, let's follow the existing decision for now.